modules/adserveradtargetBidAdapter.js

```diff
@@ -0,0 +1,408 @@
+
+const INTEGRATION_METHOD = 'prebid.js';
+const BIDDER_CODE = 'adserveradtarget';
+const ENDPOINT = 'https://z.cdn.adtarget.market/ssp?prebid&s=';
+const ADAPTER_VERSION = '1.0.0';
+const PREBID_VERSION = '$prebid.version$';
+const DEFAULT_BID_TTL = 300;
+const DEFAULT_CURRENCY = 'USD';
+
+function getSize(size) {
+  return {
+    w: parseInt(size[0]),
+    h: parseInt(size[1])
+  }
+}
+
+function transformSizes(sizes) {
+  if (isArray(sizes) && sizes.length === 2 && !isArray(sizes[0])) {
+    return [ getSize(sizes) ];
+  }
+  return sizes.map(getSize);
+}
+
+function extractUserSyncUrls(syncOptions, pixels) {
+  let itemsRegExp = /(img|iframe)[\s\S]*?src\s*=\s*("|')(.*?)\2/gi;
+  let tagNameRegExp = /\w*(?=\s)/;
+  let srcRegExp = /src=("|')(.*?)\1/;
+  let userSyncObjects = [];
+
+  if (pixels) {
+    let matchedItems = pixels.match(itemsRegExp);
+    if (matchedItems) {
+      matchedItems.forEach(item => {
+        let tagName = item.match(tagNameRegExp)[0];
+        let url = item.match(srcRegExp)[2];
+
+        if (tagName && url) {
+          let tagType = tagName.toLowerCase() === 'img' ? 'image' : 'iframe';
+          if ((!syncOptions.iframeEnabled && tagType === 'iframe') ||
+                (!syncOptions.pixelEnabled && tagType === 'image')) {
+            return;
+          }
+          userSyncObjects.push({
+            type: tagType,
+            url: url
+          });
+        }
+      });
+    }
+  }
+
+  return userSyncObjects;
+}
+
+function isSecure(bid) {
+  return deepAccess(bid, 'params.bidOverride.imp.secure') || (document.location.protocol === 'https:') ? 1 : 0;
+};
+
+function getMediaType(bid) {
+  return deepAccess(bid, 'mediaTypes.banner') ? BANNER : false;
+}
+
+function validateAppendObject(validationType, allowedKeys, inputObject, appendToObject) {
+  const outputObject = {
+    ...appendToObject
+  };
+
+  for (const objectKey in inputObject) {
+    switch (validationType) {
+      case 'string':
+        if (allowedKeys.indexOf(objectKey) !== -1 && isStr(inputObject[objectKey])) {
+          outputObject[objectKey] = inputObject[objectKey];
+        };
+        break;
+      case 'number':
+        if (allowedKeys.indexOf(objectKey) !== -1 && isNumber(inputObject[objectKey])) {
+          outputObject[objectKey] = inputObject[objectKey];
+        };
+        break;
+
+      case 'array':
+        if (allowedKeys.indexOf(objectKey) !== -1 && isArray(inputObject[objectKey])) {
+          outputObject[objectKey] = inputObject[objectKey];
+        };
+        break;
+      case 'object':
+        if (allowedKeys.indexOf(objectKey) !== -1 && isPlainObject(inputObject[objectKey])) {
+          outputObject[objectKey] = inputObject[objectKey];
+        };
+        break;
+      case 'objectAllKeys':
+        if (isPlainObject(inputObject)) {
+          outputObject[objectKey] = inputObject[objectKey];
+        };
+        break;
+    };
+  };
+  return outputObject;
```

Comment:
Instead of using a switch statement and passing the validation type, could you simply pass the validating function and in the case of objectAllKeys pass an empty array as allowedKeys (although not sure you need this case as it is never used in your adapter)? Something like this to reduce the size and repetition in this function: 

function validateAppendObject(validationFunction, allowedKeys, inputObject, appendToObject) {
   const outputObject = {
      ...appendToObject
   };

    for (const objectKey in inputObject) {
        const hasKeys = allowedKeys.length;
        const isValid = (!hasKeys && validationFunction(inputObject)) || 
            (hasKeys && allowedKeys.indexOf(objectKey) !== -1 && validationFunction(inputObject[objectKey]);
       if (isValid) outputObject[objectKey] = inputObject[objectKey];
    }
}